hail/python/hail/backend/service_backend.py

```diff
@@ -325,43 +325,50 @@ async def _rpc(self,
             with timings.step("submit batch"):
                 batch_attributes = self.batch_attributes
                 if 'name' not in batch_attributes:
-                    batch_attributes = {**batch_attributes, 'name': self.name_prefix + name}
-                bb = self.async_bc.create_batch(token=token, attributes=batch_attributes)
+                    batch_attributes = {**batch_attributes, 'name': self.name_prefix}
+                if self._batch is None:
+                    bb = self.async_bc.create_batch(token=token, attributes=batch_attributes)
+                else:
+                    bb = await self.async_bc.update_batch(self._batch)
 
                 resources: Dict[str, Union[str, bool]] = {'preemptible': False}
                 if self.driver_cores is not None:
                     resources['cpu'] = str(self.driver_cores)
                 if self.driver_memory is not None:
                     resources['memory'] = str(self.driver_memory)
 
-                bb.create_jvm_job(
+                j = bb.create_jvm_job(
                     jar_spec=self.jar_spec.to_dict(),
                     argv=[
                         ServiceBackend.DRIVER,
-                        batch_attributes['name'],
+                        batch_attributes['name'] + name,
                         iodir + '/in',
                         iodir + '/out',
                     ],
                     mount_tokens=True,
                     resources=resources,
-                    attributes={'name': 'driver'},
+                    attributes={'name': name + '_driver'},
                 )
-                b = await bb.submit(disable_progress_bar=True)
+                self._batch = await bb.submit(disable_progress_bar=True)
 
-            with timings.step("wait batch"):
+            with timings.step("wait driver"):
                 try:
                     if self.disable_progress_bar is not True:
                         deploy_config = get_deploy_config()
-                        url = deploy_config.external_url('batch', f'/batches/{b.id}/jobs/1')
-                        print(f'Submitted batch {b.id}, see {url}')
-
-                    await b.wait(description=name,
-                                 disable_progress_bar=self.disable_progress_bar,
-                                 progress=progress)
+                        # FIXME Should we not `print`?
+                        url = deploy_config.external_url('batch', f'/batches/{self._batch.id}/jobs/{j.job_id}')
+                        print(f'Submitted to batch {self._batch.id}, see {url}')
```

Comment:
I'm somewhat inclined to not print. The progress bar should provide all necessary information

Reply:
Happy to axe that.